os/drivers/serial/Kconfig

```diff
@@ -3,6 +3,27 @@
 # see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
 #
 
+config DYNAMIC_LOGS_STORING
+	bool "dynamic storing of logs"
+	default n
+	depends on DEBUG_MM_HEAPINFO
+
+config LOG_CHUNK_SIZE
+	int "dynamic log chunk size"
+	range 0 MAX_LOG_SIZE
+	default 1024
+	depends on DYNAMIC_LOGS_STORING
+
+config MAX_LOG_SIZE
+	int "dynamic log maximum size"
+	default 256000
+	depends on DYNAMIC_LOGS_STORING
+
+config MAX_FREE_HEAP_PERCENT
+	int "maximum percent of logs compared to free heap"
+	default 20
+	depends on DYNAMIC_LOGS_STORING
```

Comment:
Instead of adding `depends on DYNAMIC_LOGS_STORING` for each config you could use if condition as follows
`if depends on DYNAMIC_LOGS_STORING`
`config xxx`
`config yyy`
`endif`